packages/ripple-keypairs/src/signing-schemes/ed25519/index.ts

```diff
@@ -0,0 +1,56 @@
+import { ed25519 as nobleEd25519 } from '@noble/curves/ed25519'
+import { bytesToHex } from '@xrplf/isomorphic/utils'
+
+import type { HexString, SigningScheme } from '../../types'
+import assert from '../../utils/assert'
+import Sha512 from '../../utils/Sha512'
+
+const ED_PREFIX = 'ED'
+
+const ed25519: SigningScheme = {
+  deriveKeypair(entropy: Uint8Array): {
+    privateKey: string
+    publicKey: string
+  } {
+    const rawPrivateKey = Sha512.half(entropy)
+    const privateKey = ED_PREFIX + bytesToHex(rawPrivateKey)
+    const publicKey =
+      ED_PREFIX + bytesToHex(nobleEd25519.getPublicKey(rawPrivateKey))
+    return { privateKey, publicKey }
+  },
+
+  sign(message: Uint8Array, privateKey: HexString): string {
+    assert.ok(message instanceof Uint8Array, 'message must be array of octets')
+    assert.ok(
+      privateKey.length === 66,
+      'private key must be 33 bytes including prefix',
+    )
+    return bytesToHex(nobleEd25519.sign(message, privateKey.slice(2)))
+  },
+
+  verify(
+    message: Uint8Array,
+    signature: HexString,
+    publicKey: string,
+  ): boolean {
+    // Unlikely to be triggered as these are internal and guarded by getAlgorithmFromKey
+    assert.ok(
+      publicKey.length === 66,
+      'public key must be 33 bytes including prefix',
+    )
+    return nobleEd25519.verify(
+      signature,
+      new Uint8Array(message),
```

Comment:
Why is this copied instead of used directly?

Reply:
Maybe it is to fully ensure it is a Uint8Array and not any other buffer construct

Reply:
fixed